### PR TITLE
Unauthored attrs changes

### PR DESCRIPTION
**Pull request by feldstj**

### CODE REVIEW

**Review by ppt-adsk — DISMISSED** [review]

**Review by ppt-adsk — APPROVED** [review]

File: test/lib/ufe/testAttribute.py [hunk 1]
```diff
@@ -358,6 +358,7 @@ def testAttributeEnumString(self):
         # Use our engine method to run the bulk of the test (all the stuff from
         # the Attribute base class). We use the visibility attribute which is
         # an EnumString type.
+        import pdb ; pdb.set_trace()
```
Comment (ppt-adsk) [inline test/lib/ufe/testAttribute.py:361]: Debugging code, please remove.

**Review by ppt-adsk — APPROVED** [review]

File: lib/mayaUsd/ufe/UsdAttribute.h [hunk 2]
```diff
@@ -71,8 +75,20 @@ namespace ufe {
...
+    ~UsdAttribute() = default;
+
+    inline bool isAuthored() const { return isValid() && fUsdAttr.IsAuthored(); }
+    inline bool isValid() const { return fUsdAttr.IsValid(); }
+    std::string isEditAllowedMsg() const;
+    inline bool isEditAllowed() const { return isEditAllowedMsg().empty(); }
+    std::string typeName() const;
+    std::string defaultValue() const;
+    bool        get(PXR_NS::VtValue& value, PXR_NS::UsdTimeCode time) const;
+    bool        set(const PXR_NS::VtValue& value, PXR_NS::UsdTimeCode time);
```
Comment (hodoulp) [inline lib/mayaUsd/ufe/UsdAttribute.h:91]: Should all these new methods be protected by UFE_V4_FEATURES_AVAILABLE?

File: lib/mayaUsd/ufe/UsdAttribute.h [hunk 3]
```diff
@@ -71,8 +75,20 @@ namespace ufe {
...
 public:
-    UsdAttribute(const UsdSceneItem::Ptr& item, const PXR_NS::UsdAttribute& usdAttr);
-    ~UsdAttribute();
+#ifdef UFE_V4_FEATURES_AVAILABLE
+    UsdAttribute(const PXR_NS::UsdPrim& prim, const Ufe::AttributeDef::ConstPtr& attrDef);
+#endif
+    UsdAttribute(const PXR_NS::UsdAttribute& usdAttr);
+    ~UsdAttribute() = default;
+
+    inline bool isAuthored() const { return isValid() && fUsdAttr.IsAuthored(); }
```
Comment (hodoulp) [inline lib/mayaUsd/ufe/UsdAttribute.h:84]: Should all the following methods be defined in `UFE_ATTRIBUTE_OVERRIDES` ?

**Review by hodoulp — COMMENTED** [review]
Last minute findings.

**Pull request merged**